File: ASMR.Core/Entities/Transaction.cs
//
// asmr: Coffee Beans Management Solution
// © 2021 Pandora Karya Digital. All right reserved.
//
// Written by Danang Galuh Tegar Prasetyo [connect@danang.id]
//
// Transaction.cs
//

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using ASMR.Core.Enumerations;
using ASMR.Core.Generic;

namespace ASMR.Core.Entities;

public class Transaction : DefaultAbstractEntity
{
	public IEnumerable<TransactionItem> Items { get; set; }

	[Required] public Payment Payment { get; set; }

	[Required] [JsonIgnore] public string PaymentId { get; set; }

	[Required] public TransactionStatus Status { get; set; }

	[Required] [JsonIgnore] public User User { get; set; }

	[Required] public string UserId { get; set; }
}

File: ASMR.Web/ClientApp/src/core/entities/Transaction.ts
import EntityBase from "asmr/core/common/EntityBase"
import TransactionItem from "asmr/core/entities/TransactionItem"
import TransactionStatus from "asmr/core/enums/TransactionStatus"
import Payment from "asmr/core/entities/Payment"

interface Transaction extends EntityBase {
	items: TransactionItem[]
	payment: Payment
	status: TransactionStatus
	userId: string
}

export default Transaction
